docs/rebase.md

```diff
@@ -1,46 +1,81 @@
-    ```shell
-      git checkout  v0.10.1
-    ```
-   
-3. Create a tmp branch to do the merge work from.
-    ```shell
-      git checkout -b merge-tmp
-    ```
-   
-4. Checkout downstream master, we need to be at the top of it when we merge.
-    ```shell
-      git checkout openshift/master    
-    ```
-   
-5. Merge the tmp branch into our downstream master (note the output id from this cmd).
-    ```shell
-      echo 'merge kubernetes-sigs/external-dns v0.10.1' | git commit-tree merge-tmp^{tree} -p HEAD -p merge-tmp -F -    
-    ```
-6.  Create a new branch for the cherry-pick work.
-    ```shell
-      git branch merge-0.10.1 fa4fdf0d659cc02843a479c242bef5a6f4cbdde1 #this hash is the id from previous command, it will be different    
-    ```
-7.  Checkout the merge branch for cherry-picking.
-    ```shell
-      git checkout merge-0.10.1
-    ```    
-8. Execute the following command and note the most recent upstream commit.
-    ```shell
-      git log openshift/master
-    ```
-9. Identify the commit to be cherry-picked, from the most recent upstream commit, till the head of downstream commit. Write them into a local file for easier usage.
-    ```shell
-      git log --oneline --no-merges 35f2594745d6955625c47d7c46029cf1cb639154..openshift/master >> merges.txt
-    ```
-   
-10. Cherry-pick the commits from the file ( bottom -> up ) one by one, and resolve conflicts.
\ No newline at end of file
+# Rebasing Procedure
+
+This fork needs to be periodically synced with upstream changes. To do this we
+need to run the following steps:
+
+1. Ensure the *upstream*, *downstream* and your own fork of the repository are  
+   configured correctly as remotes.
+   ```bash
+   git remote -v
+   # if upstream or downstream are not present then add them
+   git remote add upstream https://github.com/kubernetes-sigs/external-dns
+   git remote add downstream https://github.com/openshift/external-dns
+   ```
+
+2. Update the remotes
+   ```bash
+   git remote update
+   ```
+
+3. Ensure that the `master` branch is up-to-date with the *downstream* `master`
+   branch.
+   ```bash
+   git switch master
+   git pull downstream master
+   ```
+
+4. Checkout the commit from upstream to which you want to rebase the `master`
+   branch.
+   ```bash
+   TAG_OR_COMMIT_ID="v0.14.2"
+   git checkout ${TAG_OR_COMMIT_ID}
+   git switch -c rebase-${TAG_OR_COMMIT_ID}
+   ```
+
+5. Merge the `master` branch into the current branch with
+   the [ours](https://git-scm.com/docs/merge-strategies#_merge_strategies)
+   merge strategy.
+   ```bash
+   git merge -s ours master
+   ```
+   _Note:_ This strategy should not be confused with the `ours` option for 
+   the `ort` merge strategy.
+
+7. Determine the changes present in the `master` branch which need to be
+   cherry-picked.
+   ```bash
+   git log --oneline --no-merges ${TAG_OR_COMMIT_ID}..master
+   ```
+
+8. Cherry-pick the needed changes, try to build and test the code.
```

Comment:
I think you have some implicit steps here that could be documented here:

1. Updating vendor files: since openshift tracks the vendor directory, we need to generate a carry patch for vendoring. Might be a good reminder to mention it.
2. Squashing: To keep cherry picks from getting messy, I think squashing is a good idea.
    - For example, `UPSTREAM: <carry>: openshift: Add DOWNSTREAM_OWNERS` and `UPSTREAM: <carry>: openshift: DOWNSTREAM_OWNERS - add Grant` could be squashed
    - `UPSTREAM: <carry>: openshift: Use golang 1.18 for CI builds` and `UPSTREAM: <carry>: openshift: Use golang 1.22 for CI builds` could be squashed.
3. Renaming cherry-pick commits: You do mention this below, so that fair.
    - However, I notice `upstream rebase tag documentation` and `Add .ci-operator.yaml with base_root_image` aren't labeled correctly and appeared to be carried from last time.

These are nit picks, but my thinking is more information is better for the new developer that comes along and has to rebase.

Reply:
Added some notes to this step.